fq.go

```diff
@@ -0,0 +1,183 @@
+package inflight
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"k8s.io/utils/clock"
+)
+
+// FQScheduler is a fair queuing implementation designed for the kube-apiserver.
+// FQ is designed for
+// 1) dispatching requests to be served rather than packets to be transmitted
+// 2) serving multiple requests at once
+// 3) accounting for unknown and varying service time
+type FQScheduler struct {
```

Comment:
This looks like an outdated copy of https://github.com/aaron-prindle/fq-apiserver/tree/mike-impl-v1

We may want to use package structure rather than copying code.

Reply:
Done